src/Nethermind/Nethermind.Evm/BlockExecutionContext.cs

```diff
@@ -12,12 +13,12 @@ public readonly struct BlockExecutionContext
     public BlockHeader Header { get; }
     public UInt256? BlobBaseFee { get; }
 
-    public BlockExecutionContext(BlockHeader blockHeader)
+    public BlockExecutionContext(BlockHeader blockHeader, IReleaseSpec spec)
```

Comment:
Refactor idea:
Maybe we should save the whole spec and reuse it later, while removing SpecProvider and manually getting spec in lot of places in VM or TransactionProcessor?

Reply:
sounds like a good idea, i will explore this